pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/mapper/HadoopSegmentCreationMapReduceJob.java

```diff
@@ -220,13 +221,20 @@ protected String createSegment(String dataFilePath, Schema schema, Integer seqId
       driver.build();
       // Tar the segment directory into file.
       String segmentName = driver.getSegmentName();
-      String localSegmentPath = new File(_localDiskSegmentDirectory, segmentName).getAbsolutePath();
+      File localSegmentFile = new File(_localDiskSegmentDirectory, segmentName);
+      String localSegmentPath = localSegmentFile.getAbsolutePath();
 
       String localTarPath = _localDiskSegmentTarPath + "/" + segmentName + JobConfigConstants.TARGZ;
       LOGGER.info("Trying to tar the segment to: {}", localTarPath);
       TarGzCompressionUtils.createTarGzOfDirectory(localSegmentPath, localTarPath);
       String hdfsTarPath = _localHdfsSegmentTarPath + "/" + segmentName + JobConfigConstants.TARGZ;
 
+      // Log segment size.
+      long uncompressedSegmentSize = localSegmentFile.length();
```

Comment:
If we want all the files in the directory, we might have to recursively go through the directory and get the size of all files. Unfortunately, java's length() only handles length of an individual file. We have some examples of recursive directory iteration when we gather input paths.

Reply:
Seems `FileUtils.sizeOfDirectory()` would be the best option. Thanks for pointing out! 👍